<launch>
  <arg name="mav_name" default="firefly"/>
  <arg name="frame_id" default="odom" />
  <arg name="verbose" default="false"/>
  <arg name="voxel_size" default="0.20" />
  <arg name="robot_radius" default="0.6" />
  <arg name="clear_radius" default="1.0" />

  <group ns="$(arg mav_name)">
    <node name="voxblox_node" pkg="voxblox_ros" type="esdf_server" output="screen" args="-alsologtostderr" clear_params="true">
      <remap from="pointcloud" to="vi_sensor/camera_depth/depth/points"/>
      <remap from="voxblox_node/esdf_map_out" to="esdf_map" />
      <remap from="voxblox_node/tsdf_map_out" to="tsdf_map" />

      <param name="method" value="fast" />
      <param name="publish_tsdf_map" value="true" />
      <param name="publish_esdf_map" value="true" />
      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="16" />
      <param name="esdf_max_distance_m" value="2.0" />
      <param name="max_ray_length_m" value="10.0" />
      <param name="voxel_carving_enabled" value="true" />
      <param name="color_mode" value="color" />
      <param name="use_tf_transforms" value="true" />
      <param name="update_mesh_every_n_sec" value="0.25" />
      <param name="min_time_between_msgs_sec" value="0.10" />
      <param name="clear_sphere_for_planning" value="true" />
      <param name="occupied_sphere_radius" value="4.0" />
      <param name="clear_sphere_radius" value="$(arg clear_radius)" />
      <param name="slice_level" value="1.0" />
      <param name="world_frame" value="$(arg frame_id)" />
      <param name="verbose" value="false" />
    </node>

    <node name="mav_local_planner" pkg="mav_local_planner" type="mav_local_planner_node" args="-alsologtostderr" output="screen" clear_params="true">
      <remap from="odometry" to="ground_truth/odometry" />
      <remap from="mav_local_planner/esdf_map_in" to="esdf_map" />
      <remap from="mav_local_planner/tsdf_map_in" to="tsdf_map" />

      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="16" />
      <param name="esdf_max_distance_m" value="2.0" />
      <param name="update_mesh_every_n_sec" value="0.0" />
      <param name="traversability_radius" value="$(arg robot_radius)" />
      <param name="publish_traversable" value="true" />
      <param name="slice_level" value="1.0" />
      <param name="publish_slices" value="true" />

      <param name="local_frame_id" value="$(arg frame_id)" />
      <param name="world_frame" value="$(arg frame_id)" />
      <param name="replan_dt" value="0.25" />
      <param name="command_publishing_dt" value="0.25" />
      <param name="replan_lookahead_sec" value="1.0" />
      <param name="mpc_prediction_horizon" value="300" />

      <param name="robot_radius" value="$(arg robot_radius)" />
      <param name="planning_horizon_m" value="20.0" />
      <param name="autostart" value="true" />
      <param name="verbose" value="true" />
      <param name="v_max" value="2.0" />
      <param name="a_max" value="2.0" />
      <param name="avoid_collisions" value="true" />

      <!-- What to do in case of failure. -->
      <param name="goal_selector_strategy" value="none" />
    </node>
  </group>
</launch>
